src/plugins/intel_cpu/thirdparty/llmdnn/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cmake_minimum_required(VERSION 3.13)
+
+project(root)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+option(LLMDNN_BUILD_TESTS "Build with tests" ON)
+
+message(INFO "--------------------------------")
+message(STATUS "Build with tests: ${LLMDNN_BUILD_TESTS}")
+message(INFO "--------------------------------")
+
+set(CMAKE_CXX_STANDARD 11)
+if(MSVC)
+  # TODO: validate
+  if(MSVC_VERSION VERSION_LESS 1928)
+    message(FATAL_ERROR "Insufficient msvc compiler version, current ${MSVC_VERSION}, minimum 1928.")
+  endif()
+  # Force to always compile with W4
+  if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+    string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4")
+  endif()
+elseif(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11.3")
+    message(FATAL_ERROR "Insufficient gcc compiler version, current ${CMAKE_CXX_COMPILER_VERSION}, minimum 11.3.")
+  endif()
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=sapphirerapids")
```

Comment:
do we use runtime check before usage of this library? Otherwise we can have illegal instructions.

Reply:
At runtime there is already checked the native [ISA](https://github.com/openvinotoolkit/openvino/blob/faa0c79edebb4715d42d6c3508fdd84c2f6ac34e/src/plugins/intel_cpu/src/nodes/fullyconnected.cpp#L937), for compiling stage maybe we should allow any machine to compile the library successfully using appropriate compiler.